kotest-extensions/src/jvmMain/kotlin/io/kotest/extensions/locale/TimezoneExtensions.kt

```diff
@@ -58,6 +59,30 @@ class TimeZoneTestListener(timeZone: TimeZone) : TimeZoneListener(timeZone), Tes
    override suspend fun afterTest(testCase: TestCase, result: TestResult) {
       resetTimeZone()
    }
+
+   override suspend fun beforeContainer(testCase: TestCase) {
+      if (testCase.type == TestType.Container) changeTimeZone()
+   }
+
+   override suspend fun afterContainer(testCase: TestCase, result: TestResult) {
+      if (testCase.type == TestType.Container) resetTimeZone()
+   }
+
+   override suspend fun beforeEach(testCase: TestCase) {
+      if (testCase.type == TestType.Test) changeTimeZone()
```

Comment:
We probably only need to call resetTimeZone /  changeTimeZone on one of the beforeAny / beforeTest cases as that will cover it for every test.